R/iSEE-main.R

```diff
@@ -498,86 +498,86 @@ iSEE <- function(se,
         enc_names <- rep(names(pObjects$memory), n_available)
         available_panels <- paste0(enc_names, enc_ids)
         names(available_panels) <- .decode_panel_name(enc_names, enc_ids)
-
+        
         # Panel ordering, addition and deletion.
         observeEvent(input$organize_panels, {
             active_panels <- paste0(rObjects$active_panels$Type, rObjects$active_panels$ID)
             names(active_panels) <- .decode_panel_name(rObjects$active_panels$Type, rObjects$active_panels$ID)
-
-            ordered_panel_choices <- c(
-                active_panels,
-                setdiff(available_panels, active_panels)
-            )
+            inactive_panels <- available_panels[which(!available_panels %in% active_panels)]
+            ordered_panel_choices <- c(active_panels, inactive_panels)
             showModal(modalDialog(
                 title = "Panel organization", size = "m", fade = TRUE,
                 footer = NULL, easyClose = TRUE,
+                actionButton("update_ui", "Apply settings", icon=icon("object-ungroup"), width = '100%'),
```

Comment:
This is why I originally did not put the selectize under the control of the action button, (this way adding/moving/removing in the selectize takes immediate effect both in the modal (add/move/remover inputs) and the UI panels. 
However, @csoneson pointed out:
> just playing with the panel width/height controls. This is what I was doing:
> (1) Change the width of all panels (nothing happened since I didn't press "Apply settings")
> (2) Change the order of two panels. This immediately moved the panels, and the width modifications from step (1) were lost.
> This seems unintentional.

It was intentional, but I suggested that:
> We can leave it to the user to:
> 1. reorder panels in the selectize (at this point, the order of panels in the selectize and width/height inputs will not match)
> 2. apply the changes to reorder both panels and height/width inputs to match the selectize
> 3. change width/height ad libitum
> 4. apply the second round of changes

It's easy enough to switch between the two behaviours, we just need to agree which behaviour is desired.

Reply:
See my suggestion on Slack to provide unsurprising behaviour. The idea would be to have an additional reactive variable (`working_initialPanels`, if you will). This gets updated with the current `initialPanels` when the modal is initialized. The width/height box UI is dependent on this reactive, while the selectize and width/height widgets can modify it. Once you press the button, the `working_initialPanels` then overwrites the original `initialPanels`. This effectively creates a working copy on which the modal-specific observers and widgets can act/respond to, without triggering an expensive re-render of the entire app.